depmapomics/upload.py

```diff
@@ -0,0 +1,1298 @@
+                        (prs_in_model.Datatype == "wgs")
+                        | (prs_in_model.Datatype == "wes")
+                    ]
+                )
+                == 1
+            ):
+                pr = prs_in_model[
+                    (prs_in_model.Datatype == "wgs") | (prs_in_model.Datatype == "wes")
+                ].index[0]
+                rows.append((m, pr, "dna"))
+            elif (
+                len(
+                    prs_in_model[
+                        (prs_in_model.Datatype == "wgs")
+                        | (prs_in_model.Datatype == "wes")
+                    ]
+                )
+                > 1
+            ):
+                # assuming SANGER doesn't have wgs
+                cds_ids_wgs = prs_in_model[
+                    prs_in_model.Datatype == "wgs"
+                ].CDSID.tolist()
+                cds_ids_wes = prs_in_model[
+                    (prs_in_model.Datatype == "wes") & (prs_in_model.CDSID != "")
+                ].CDSID.tolist()  # CDSID is '' when the profile is in legacy
+                pr = ""
+                # if no wgs, look at MC table and select broad wes over sanger wes
+                if len(cds_ids_wgs) == 0:
+                    if len(prs_in_model[prs_in_model.Datatype == "wes"]) == 1:
+                        pr = prs_in_model[prs_in_model.Datatype == "wes"].index[0]
+                    else:
+                        subset_seq_table = seq_table[seq_table.index.isin(cds_ids_wes)]
+                        subset_seq_table.source = subset_seq_table.source.replace(
+                            source_priority
+                        )
+                        latest_cds_id_wes = subset_seq_table.loc[
+                            cds_ids_wes, "source"
+                        ].idxmin()
+                        pr = subset_pr_table[
+                            subset_pr_table.CDSID == latest_cds_id_wes
+                        ].index[0]
+                # if there is wgs, always select wgs
+                else:
+                    subset_seq_table = seq_table[seq_table.index.isin(cds_ids_wgs)]
+                    subset_seq_table.source = subset_seq_table.source.replace(
+                        source_priority
+                    )
+                    latest_cds_id_wgs = subset_seq_table.loc[
+                        cds_ids_wgs, "source"
+                    ].idxmin()
+                    pr = subset_pr_table[
+                        subset_pr_table.CDSID == latest_cds_id_wgs
+                    ].index[0]
+                rows.append((m, pr, "dna"))
+    default_table = pd.DataFrame(rows, columns=["ModelID", "ProfileID", "ProfileType"])
+    return default_table
+
+
+def initVirtualDatasets(
+    samplesetname=SAMPLESETNAME, taiga_folder_id=VIRTUAL_FOLDER, portals=DATASETS
+):
+    """initialize taiga virtual datasets for all 4 portals by uploading an empty dummy file
+    """
+
+    with open("temp/dummy.csv", "w") as fp:
```

Comment:
I'd add a comment to explain why you're creating an empty file.

(You could also just upload /dev/null instead of creating an empty file)